src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/strategy/StrategyContainer.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.strategy;
+
+import com.google.common.collect.ImmutableSet;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Classification;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+
+/**
+ * Container class to hold availableStrategies strategies enabled by this project.
+ * In order to enable one must do:
+ * <ul>
+ *     <li>implement {@link AbstractStrategy}</li>
+ *     <li>include {@code @Resource(name = "strategy-name") private someStrategy;}</li>
+ *     <li>stick resource in {@link this#availableStrategies} via {@link this#setUpStrategies()}</li>
+ * </ul>
+ */
+@Component
+public class StrategyContainer {
+
+    private Set<Strategy> availableStrategies = Collections.emptySortedSet();
+
+    @PostConstruct
+    public void setUpStrategies() {
+        availableStrategies = new LinkedHashSet<>(ImmutableSet.of(
+        // set of strategies
+        ));
+    }
+
+    public Strategy getStrategy(Envelope envelope, CaseDetails caseDetails) {
+        return availableStrategies
```

Comment:
Given we'll have 3 or 4 strategies, I'm not sure if we need to be that generic. How about a switch or sth? This way a developer can quickly figure out what's going on.